http/transport/retry_round_tripper.go

```diff
@@ -77,17 +76,11 @@ func (l *RetryRoundTripper) SetTransport(rt http.RoundTripper) {
 // RoundTrip executes a HTTP request with retrying
 func (l *RetryRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
 	retryTransport := *l.retryTransport
-	retryTransport.Next = transportWithAttempt(l.Transport())
+	retryTransport.RoundTripper = transportWithAttempt(l.Transport())
 	resp, err := retryTransport.RoundTrip(req)
 
-	var maxRetriesError retry.MaxError
 	if err != nil {
-		switch {
-		case errors.As(err, &maxRetriesError):
-			return nil, ErrRetryFailed
-		default:
-			return nil, err
-		}
+		return nil, err
```

Comment:
We should keep mapping the `ErrRetryFailed` as code may depends on it.

Reply:
The used package does not provide a dedicated error on exceeded retry. Instead the normal request error / status is returned. Currently I do not see an easy way to achieve this.